File: frontend/src/types/save.ts
export interface Save {
  id: number;
  name: string;
  slug: string;
  description: string | null;
  accent_color: string;
  secondary_color: string;
  created_at: string;
  updated_at: string;
}

export interface SavePayload {
  name: string;
  slug?: string | null;
  description?: string | null;
  accent_color: string;
  secondary_color: string;
}

File: backend/app/models/save.py
from datetime import datetime, timezone

from sqlalchemy import DateTime, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


def utc_now() -> datetime:
    return datetime.now(timezone.utc)


class Save(Base):
    __tablename__ = "saves"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    name: Mapped[str] = mapped_column(String(120), nullable=False)
    slug: Mapped[str] = mapped_column(String(140), nullable=False, unique=True, index=True)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    accent_color: Mapped[str] = mapped_column(String(16), nullable=False, default="#ff6f61")
    secondary_color: Mapped[str] = mapped_column(String(16), nullable=False, default="#151515")
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=utc_now)
    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=utc_now, onupdate=utc_now)
